### ISSUE TITLE
Dependency errors:

### ISSUE THREAD

**Issue by iWhacko** [walterfr/UltraStarKaraokeMaker#9]
Hi I'm getting some dependency errors, reinstall does not fix it. I'm kinda sure it worked before, so maybe something is wrong in the latest versions, 
Also it says no language was specified, but I actually did.

`──── Etapa 4/6 — Alinhando letra ao áudio (WhisperX, âncora+interpolação) ─────
C:\Users\<username>\AppData\Local\USKMaker\venv\Lib\site-packages\pyannote\audio\core\io.py:48: UserWarning: 
torchcodec is not installed correctly so built-in audio decoding will fail. Solutions are:
* use audio preloaded in-memory as a {'waveform': (channel, time) torch.Tensor, 'sample_rate': int} dictionary;
* fix torchcodec installation. Error message was:
Could not load libtorchcodec. Likely causes:
          1. FFmpeg is not properly installed in your environment. We support
             versions 4, 5, 6 and 7.
          2. The PyTorch version (2.8.0+cu126) is not compatible with
             this version of TorchCodec. Refer to the version compatibility
             table:
             https://github.com/pytorch/torchcodec?tab=readme-ov-file#installing-torchcodec.
          3. Another runtime dependency; see exceptions below.
        The following exceptions were raised as we tried to load libtorchcodec:
        
[start of libtorchcodec loading traceback]
FFmpeg version 7: Could not find module 'C:\Users\<username>\AppData\Local\USKMaker\venv\Lib\site-packages\torchcodec\libtorchcodec_core7.dll' (or one of its dependencies). Try using the full path with constructor syntax.
FFmpeg version 6: Could not find module 'C:\Users\<username>\AppData\Local\USKMaker\venv\Lib\site-packages\torchcodec\libtorchcodec_core6.dll' (or one of its dependencies). Try using the full path with constructor syntax.
FFmpeg version 5: Could not find module 'C:\Users\<username>\AppData\Local\USKMaker\venv\Lib\site-packages\torchcodec\libtorchcodec_core5.dll' (or one of its dependencies). Try using the full path with constructor syntax.
FFmpeg version 4: Could not find module 'C:\Users\<username>\AppData\Local\USKMaker\venv\Lib\site-packages\torchcodec\libtorchcodec_core4.dll' (or one of its dependencies). Try using the full path with constructor syntax.
[end of libtorchcodec loading traceback].
  warnings.warn(
2026-07-28 15:12:35 - whisperx.asr - INFO - No language specified, language will be detected for each audio file (increases inference time)
2026-07-28 15:12:35 - whisperx.vads.pyannote - INFO - Performing voice activity detection using Pyannote...
Lightning automatically upgraded your loaded checkpoint from v1.5.4 to v2.6.5. To apply the upgrade to your files permanently, run `python -m lightning.pytorch.utilities.upgrade_checkpoint C:\Users\<username>\AppData\Local\USKMaker\venv\Lib\site-packages\whisperx\assets\pytorch_model.bin`
C:\Users\<username>\AppData\Local\USKMaker\venv\Lib\site-packages\pyannote\audio\utils\reproducibility.py:74: ReproducibilityWarning: TensorFloat-32 (TF32) has been disabled as it might lead to reproducibility issues and lower accuracy.
It can be re-enabled by calling
   >>> import torch
   >>> torch.backends.cuda.matmul.allow_tf32 = True
   >>> torch.backends.cudnn.allow_tf32 = True
See https://github.com/pyannote/pyannote-audio/issues/1370 for more details.
  warnings.warn(
[INFO] Realinhamento de janela: 37 palavras medidas no 2º passe.
OK 301 palavras processadas.
    222 âncora exata / 32 fuzzy / 37 realinhadas no 2º passe / 0 início de 
linha (.lrc) / 10 interpoladas (estimadas)
AVISO 3.3% das palavras ficaram interpoladas (não foi possível medi-las no 
áudio, nem no 2º passe) - maiores sequências seguidas: [5, 4, 1].
AVISO A letra fornecida termina em 221.8s, mas o áudio tem 241.9s (20.1s sem 
nenhuma palavra no final). Isso costuma acontecer quando um refrão/trecho 
repetido foi escrito só uma vez na letra (ex.: letras de sites que usam 
"(2x)"/"(4x)" em vez de repetir o texto por extenso). Se for o caso, reescreva 
a letra repetindo o trecho tantas vezes quanto ele é cantado.
───────────── Etapa 5/6 — Buscando metadados (capa, ano, gênero) ──────────────`

**Comment by walterfr** [comment]
Thanks for the report! A couple things:

1. The "torchcodec" warning is known noise — it falls back to the standard decoder, doesn't affect the result. Safe to ignore.
2. The "no language specified" message was a confusing log on our side, not an actual bug — the language you picked was always correctly used for transcription. Fixed the log for the next version, so it won't show up anymore.
3. Looking at the log you pasted, the alignment actually came out **good** (97% of words measured, only 3.3% estimated) — no failure there. It looks like it either got stuck or the paste just got cut off after "Etapa 5/6". Could you share what happens **after** that step? Did the package end up getting generated, or does it fail there?

**Comment by iWhacko** [comment]
Oh it created everything just fine, but I have trouble with some songs only getting 50% or so. and since I saw the errors, I thought maybe it wasn't detecting the workds because it didn't use the right decoder.

**Comment by walterfr** [comment]
Good news that the package came out! About the "50% or less" thing: it's most likely **not** the torchcodec fallback — the fallback decoder works the same, it's just slower/more verbose in the log. The more common real cause: Whisper (the recognition AI) struggles with certain vocal styles — heavily processed/electronic, screamed, whispered, mixed low in the track — regardless of which decoder is used. It's a known limitation of the tool itself, not a one-off bug.

A couple things already in the app for exactly this case:

1. **Low-confidence warning banner** on the result screen (made a lot more visible in v0.13.0) — flags it when this happens.
2. **"Generate again"** — vocal separation varies each run, sometimes the 2nd attempt does better.
3. **Review screen** — fix just the bad parts without regenerating everything.

If you want, send me the name of one of the songs that got ~50% and I can look into whether it's this same limitation or something specific to that track.

**Comment by walterfr** [comment]
Confirmed by the log — thanks for sending it! `WORD-RECALL BAIXO: 37%` matches your suspicion exactly: this is the same "Boy Harsher - Pain" from your Reddit post, and I tested free transcription on it — Whisper barely recognizes anything in that vocal. Deep/processed darkwave vocals are quite far from what the tool was trained to understand.

It's not a one-off bug — it's a real ceiling of the recognition AI for this kind of vocal. We already investigated several architecture alternatives in a large session, none of them fix this case. No quick fix available today, unfortunately.

The final ~60s with no notes at all is most likely instrumental beyond what your lyrics cover — normal, not a failure.

Thanks for documenting it so well — it confirms exactly where the tool's current limit is.

**Comment by iWhacko** [comment]
I didnt post that one on reddit, the one i'm having trouble with is this one: https://www.youtube.com/watch?v=OIfmQ6BYP_Q
it sometimes gets really close though. and then screws up some timings, and the editor makes it hard to move multiple words on the time line, selecting multiple words, or inserting a space a a certain point would be great. I might make an editor myself later, but havent had the time for it yet

**Comment by walterfr** [comment]
Follow-up: dug into "Why Do I Have To Feel?" (the one you linked) and found something different from what I first thought.

The vocal recognition itself is actually fine on this one when it tries — but the voice-activity detector was skipping ~70% of the track entirely (quiet/atmospheric vocals under the mix don't cross its default threshold), so most of the song had nothing to anchor to at all.

Just shipped v0.16.0 with an automatic rescue for exactly this: when alignment comes out bad, the app now also tries a more sensitive voice detector and keeps whichever result is actually better. Measured against 51 independent charts first to make sure it doesn't regress songs that already work (it's conditional, only kicks in when things are already going poorly).

Also shipped in this release: you can now paste a `.lrc` file's contents directly if the built-in search comes up empty (common for less-indexed genres) — no need to work around it manually anymore.

Update to the latest version and let me know if it gets better on your end!
